code/game/objects/items/weapons/storage/book.dm

```diff
@@ -79,22 +80,16 @@
 			ticker.mode.remove_cultist(M.mind)*/
 		if ((istype(M, /mob/living/carbon/human) && prob(60)))
 			bless(M)
-			if(ishuman(M))
-				var/mob/living/carbon/human/H = M
+			if(bless(M))
 				var/message_halt = 0
-				for(var/obj/item/organ/limb/affecting in H.organs)
-					if(affecting.status == ORGAN_ORGANIC)
-						if(message_halt == 0)
-							M.visible_message("<span class='notice'>[user] heals [M] with the power of [src.deity_name]!</span>")
-							M << "<span class='boldnotice'>May the power of [src.deity_name] compel you to be healed!</span>"
-							playsound(src.loc, "punch", 25, 1, -1)
-							message_halt = 1
-					else
-						user << "<span class='warning'>[src.deity_name] refuses to heal this metallic taint!</span>"
-						return
-
-
-
+				if(message_halt == 0)
+					M.visible_message("<span class='notice'>[user] heals [M] with the power of [src.deity_name]!</span>")
+					M << "<span class='boldnotice'>May the power of [src.deity_name] compel you to be healed!</span>"
+					playsound(src.loc, "punch", 25, 1, -1)
+					message_halt = 1
```

Comment:
Removed message_halt and fixed the loop ending early.

On 3 January 2015 at 17:46, phil235 <notifications@github.com> wrote:

> In code/game/objects/items/weapons/storage/book.dm
> <https://github.com/tgstation/-tg-station/pull/6775#discussion-diff-22433628>
> :
>
> > -						if(message_halt == 0)
> > -							M.visible_message("<span class='notice'>[user] heals [M] with the power of [src.deity_name]!</span>")
> > -							M << "<span class='boldnotice'>May the power of [src.deity_name] compel you to be healed!</span>"
> > -							playsound(src.loc, "punch", 25, 1, -1)
> > -							message_halt = 1
> > -					else
> > -						user << "<span class='warning'>[src.deity_name] refuses to heal this metallic taint!</span>"
> > -						return
> > -
> > -
> > -
> > +				if(message_halt == 0)
> > +					M.visible_message("<span class='notice'>[user] heals [M] with the power of [src.deity_name]!</span>")
> > +					M << "<span class='boldnotice'>May the power of [src.deity_name] compel you to be healed!</span>"
> > +					playsound(src.loc, "punch", 25, 1, -1)
> > +					message_halt = 1
>
> The message_halt var and check is no longer needed now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/6775/files#r22433628>.
>



-- 
Matt Smith, Riot Games
----------------------------